lib/gems/pending/util/mount/miq_s3_session.rb

```diff
@@ -0,0 +1,63 @@
+require 'util/mount/miq_generic_mount_session'
+require 'aws-sdk'
+
+class MiqS3Session < MiqGenericMountSession
+  def initialize(log_settings)
+    log_header = "MIQ(#{self.class.name}-initialize)"
+    logger.debug("#{log_header} initialize: log_settings are #{log_settings}")
+    super(log_settings)
+    logger.debug("#{log_header} initialize: @settings are #{@settings}")
+    raise "username is a required value!" if @settings[:username].nil?
+    raise "password is a required value!" if @settings[:password].nil?
+    raise "region is a required value!" if @settings[:region].nil?
+    @s3 = s3
+  end
+
+  def s3
+    @s3 ||= Aws::S3::Resource.new(:region => @settings[:region], :access_key_id => @settings[:username], :secret_access_key => @settings[:password])
+  end
+
+  def connect
+    _scheme, _userinfo, @host, _port, _registry, @mount_root, _opaque, _query, _fragment = URI.split(URI.encode(@settings[:uri]))
+    @mount_path = @mount_root.split("/")[0..1].join("/")
```

Comment:
This seems strange. What is it doing?

Reply:
Removing the URI prefix.  (s3://).  The way the UI is implemented at this point the part of the prefix prior to the "://" is what determines the file_depot subclass. I'm not saying that's the best way...

Reply:
Then you should `split("://")` and take the second field - no need to join.
Here again, look into the `URI` facility, I think it will accomplish this more cleanly.

Reply:
Yup.  I'm reworking this all using URI.

Reply:
Can I suggest `Pathname.new(@mount_root).each_filename.first`?

```ruby
irb(main):023:0> _scheme, _userinfo, @host, _port, _registry, @mount_root, _opaque, _query, _fragment = URI.split(URI.encode("s3://host/some/path/goes/here"))
=> ["s3", nil, "host", nil, nil, "/some/path/goes/here", nil, nil, nil]
irb(main):024:0> Pathname.new(@mount_root).each_filename.first
=> "some"
```

Reply:
I'm not sure that's exactly your desired behavior, but a spec would probable help too :smile: